include/xtensor/xbuilder.hpp

```diff
@@ -237,26 +242,31 @@ namespace xt
             template <class It>
             inline R element(It first, It) const
             {
-                // Avoids warning when T = char (because char + char => int!)
-                using mult_type = get_mult_type<T, S>;
-                return static_cast<R>(m_start + m_step * mult_type(*first));
+                return access_impl(*first);
             }
 
             template <class E>
             inline void assign_to(xexpression<E>& e) const noexcept
             {
-                arange_assign_to<R>(e, m_start, m_step);
+                arange_assign_to<R>(e, m_start, m_stop, m_step, m_endpoint);
             }
 
         private:
 
             T m_start;
             T m_stop;
             step_type m_step;
+            size_t m_num_steps;
+            bool m_endpoint; // true for setting the last element to m_stop
 
             template <class T1, class... Args>
             inline R access_impl(T1 t, Args...) const
             {
+                if (m_endpoint && m_num_steps > 1 && size_t(t) == m_num_steps - 1)
```

Comment:
I don't see an easy way to check the last point unless introducing another member `m_num_steps` to the class `arange_generator`.

Reply:
I don't see another way either, and I thnk it's fine to add this member.